elfpy/markets/hyperdrive/hyperdrive_pricing_model_sol.py

```diff
@@ -46,8 +44,9 @@ def calculate_spot_price(
     FixedPoint
         The spot price of bonds in terms of base as an 18 fixed-point value.
     """
-    spot_price = (initial_share_price * share_reserves / bond_reserves) ** time_stretch
-    return spot_price
+    if bond_reserves <= FixedPoint(0):
```

Comment:
can't land this.  any file that ends in sol.py ought to mirror the solidity, per the function comment above.  move this up a level.